sqlglot/tokens.py

```diff
@@ -1046,12 +1051,25 @@ def _scan_formatted_string(self, string_start: str) -> bool:
         return True
 
     def _scan_identifier(self, identifier_end: str) -> None:
-        while self._peek != identifier_end:
+        text = ""
+        identifier_end_is_escape = identifier_end in self._IDENTIFIER_ESCAPES
+
+        while True:
             if self._end:
                 raise RuntimeError(f"Missing {identifier_end} from {self._line}:{self._start}")
+
             self._advance()
-        self._advance()
-        self._add(TokenType.IDENTIFIER, self._text[1:-1])
+            if self._char == identifier_end:
+                if identifier_end_is_escape and self._peek == identifier_end:
+                    text += 2 * identifier_end  # type: ignore
```

Comment:
why are you doing 2 * identifier_end, shouldn't you use escape?

Reply:
Yeah, I think you're right. I wanted to make it so transpilation yields the same string, but we can probably do that by expanding a single `"` into two inside the generator.